### PR TITLE
#1040 Support kind and size options in action buttons

### LINKED ISSUE #1040: Support `kind` and `size` options in action buttons that would map to the same props as carbon buttons
Carbon buttons documentation - https://v10-react.carbondesignsystem.com/?path=/docs/components-button--default

- Button kinds - `primary`, `secondary`, `tertiary`, `ghost`, `danger`, `danger--tertiary`, and `danger--ghost`
- Default button kind - `tertiary`
- Button sizes - `sm`, `md`, `lg`, `xl`
- Default button size - `sm`

Added a new config option in uiHints `action_info` - 
```
"button": {
          "kind": "secondary",
          "size": "lg"
 }
```

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes #1040 

Screen Shot 2022-04-29 at 10 59 10 AM

Screen Shot 2022-04-29 at 10 59 17 AM

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by matthoward366** [comment]
I know this isn't new but can we fix the tooltip on these buttons.  
Screen Shot 2022-05-03 at 4 12 03 PM

I'm not sure if we need something for this case.  There should be some padding under the button.
Screen Shot 2022-05-03 at 4 14 31 PM

**Comment by nmgokhale** [comment]
Fixed button tooltip and added 16px bottom padding for action buttons under control-items.

File: canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx [hunk 1]
```diff
@@ -32,6 +33,35 @@ class ButtonAction extends React.Component {
...
+	getActionButtonSize() {
+		if (!has(this.props, "action.button.size")) {
+			return CARBON_BUTTON_SIZE.SMALL;
+		}
+		switch (this.props.action.button.size) {
+		case CARBON_BUTTON_SIZE.SMALL: return CARBON_BUTTON_SIZE.SMALL;
+		case CARBON_BUTTON_SIZE.MEDIUM: return CARBON_BUTTON_SIZE.DEFAULT; // TODO: update this after upgrading to carbon 11.x
+		case CARBON_BUTTON_SIZE.LARGE: return CARBON_BUTTON_SIZE.LARGE;
+		case CARBON_BUTTON_SIZE.EXTRA_LARGE: return CARBON_BUTTON_SIZE.EXTRA_LARGE;
+		default: return CARBON_BUTTON_SIZE.SMALL;
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx:61]: Shouldn't this be `medium` to match the default from before?

Comment (nmgokhale) [reply canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx:61]: Default size is `small` - https://github.com/elyra-ai/pipeline-schemas/blob/master/common-pipeline/operators/uihints-v3-schema.json#L628

Default `small` is also returned on line 54.

Comment (matthoward366) [reply canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx:61]: Wasn't the default md before though?  Maybe we need to update the schema.

**Review by matthoward366 — APPROVED** [review]

**Pull request merged**